### PR TITLE
[wasm] Correctly set PATH for node, and python

### LINKED ISSUE #51535: WasmApp.targets expects python to be in the path to run
SDk v6.0.100-preview.4.21219.4

The preview4 SDK expects a compatible version of python to be installed in the path and does not use the python from the bundled pack. Installing python manually works around the issue.

### PR DESCRIPTION
**Pull request by radical**
This showed up on windows, if you didn't have `python` in PATH. Set this
correctly, by using the relevant properties from the packages, and
setting fallbacks in case of EMSDK_PATH being used.

Fixes https://github.com/dotnet/runtime/issues/51535 .

### CODE REVIEW

**Comment by radical** [comment]
This depends on the updated packages after https://github.com/dotnet/emsdk/pull/10 is merged.

File: src/mono/wasm/build/WasmApp.targets [hunk 1]
```diff
@@ -256,8 +258,10 @@
       <_PythonToolsBasePath>$(EmscriptenSdkToolsPath)python</_PythonToolsBasePath>
       <_PythonToolsVersionedPath Condition="Exists($(_PythonToolsBasePath))">$([System.IO.Directory]::GetDirectories($(_PythonToolsBasePath)))</_PythonToolsVersionedPath>
       <EmscriptenPythonToolsPath Condition="'$(_PythonToolsVersionedPath)' != ''">$(_PythonToolsVersionedPath)</EmscriptenPythonToolsPath>
+      <EmscriptenPythonBinPath   Condition="'$(EmscriptenPythonToolsPath)' != ''">$(EmscriptenPythonToolsPath)\bin\</EmscriptenPythonBinPath>
```
Comment (lewing) [inline src/mono/wasm/build/WasmApp.targets:261]: Is this correct?

Comment (radical) [reply src/mono/wasm/build/WasmApp.targets:261]: It is, when using emsdk checkout, like `src/mono/wasm/emsdk/python/3.7.4-2_64bit/bin`

Comment (lewing) [reply src/mono/wasm/build/WasmApp.targets:261]: on windows too?

**Review by lewing — APPROVED** [review]
Looks ok to me, have we tested it on windows with the new packages?  I'd like to be very sure

File: src/mono/wasm/build/WasmApp.targets [hunk 2]
```diff
@@ -273,7 +278,9 @@
     <PropertyGroup>
       <EmscriptenSdkToolsPath    Condition="'$(EmscriptenSdkToolsPath)' != ''"   >$([MSBuild]::NormalizeDirectory($(EmscriptenSdkToolsPath)))</EmscriptenSdkToolsPath>
       <EmscriptenNodeToolsPath   Condition="'$(EmscriptenNodeToolsPath)' != ''"  >$([MSBuild]::NormalizeDirectory($(EmscriptenNodeToolsPath)))</EmscriptenNodeToolsPath>
+      <EmscriptenNodeBinPath     Condition="'$(EmscriptenNodeBinPath)' != ''"    >$([MSBuild]::NormalizeDirectory($(EmscriptenNodeBinPath)))</EmscriptenNodeBinPath>
       <EmscriptenPythonToolsPath Condition="'$(EmscriptenPythonToolsPath)' != ''">$([MSBuild]::NormalizeDirectory($(EmscriptenPythonToolsPath)))</EmscriptenPythonToolsPath>
+      <EmscriptenPythonBinPath   Condition="'$(EmscriptenPythonBinPath)' != ''"  >$([MSBuild]::NormalizeDirectory($(EmscriptenPythonBinPath)))</EmscriptenPythonBinPath>
       <EmSdkUpstreamBinPath      Condition="'$(EmSdkUpstreamBinPath)' != ''"     >$([MSBuild]::NormalizeDirectory($(EmSdkUpstreamBinPath)))</EmSdkUpstreamBinPath>
```
Comment (lewing) [inline src/mono/wasm/build/WasmApp.targets:284]: nit : why EmSdkUpstreamBinPath  vs EmscriptenSdkUpstreamBinPath ?

File: src/mono/wasm/build/WasmApp.targets (hunk 1 above)
Comment (radekdoulik) [reply src/mono/wasm/build/WasmApp.targets]: on windows I have `src\mono\wasm\emsdk\python\3.7.4-pywin32_64bit\python.exe`

**Review by radekdoulik — APPROVED** [review]
LGTM.

Not sure how to properly test it on Windows. @lewing, I think you mentioned there's now a way to use the custom packages. Is it documented somewhere?

**Comment by radical** [comment]
Don't merge this yet

**Comment by radical** [comment]
Corresponding `emsdk` PR https://github.com/dotnet/emsdk/pull/12 .

**Review by lewing — APPROVED** [review]
Looks ok, we really need some cross platform CI for the aot builds

File: src/mono/wasm/build/WasmApp.targets [hunk 3]
```diff
@@ -226,38 +226,13 @@
...
-
-      <_NodeToolsBasePath>$(EmscriptenSdkToolsPath)node</_NodeToolsBasePath>
-
-      <!-- gets the path like emsdk/python/3.7.4-2_64bit -->
-      <_NodeToolsVersionedPath Condition="Exists($(_NodeToolsBasePath))">$([System.IO.Directory]::GetDirectories($(_NodeToolsBasePath)))</_NodeToolsVersionedPath>
-      <EmscriptenNodeToolsPath Condition="'$(_NodeToolsVersionedPath)' != ''">$(_NodeToolsVersionedPath)</EmscriptenNodeToolsPath>
-
-      <_UsingEMSDK_PATH>true</_UsingEMSDK_PATH>
-    </PropertyGroup>
+  <Import Project="$(MSBuildThisFileDirectory)EmSdkRepo.props" Condition="'$(EMSDK_PATH)' != ''" />
```
[the remark this answers is not in the document]
Comment (radical) [reply src/mono/wasm/build/WasmApp.targets:229]: Excellent idea!

**Pull request merged**